### DISCUSSION TITLE
WTH we cannot access the "icon" attribute for entities?

### DISCUSSION THREAD

**Discussion by ildar170975** [home-assistant/architecture#931] [Entity Models]
Consider this:
![image](https://github.com/home-assistant/architecture/assets/71872483/5b729492-b322-4382-91b7-51150a20776c)

Seems that we may only access an icon if:
-- the icon is customized
-- the icon is defined explicitly - only for template sensors.
-- ... probably some other cases

Why we cannot access icons in all cases?

**Comment by thecode** [comment]
This place is used for developers to discuss architectural changes in Home Assistant, not for creating issues. I don't see anything in this discussion that requires architectural discussion. If you are a developer that want to add this to the template feel free to create a PR for that.
If you think this is a bug the right place for it is https://github.com/home-assistant/core/issues/new/choose

If you are in need of support I highly recommend to try our [Community Forum](https://community.home-assistant.io) or join our [Discord chat server](https://discord.gg/c5DvZ4e).
You are more likely to get the answer you are looking for in those places.

Thanks! 👍
